migrations/10_ribbon_yvweth_covered_call.js

```diff
@@ -0,0 +1,74 @@
+const RibbonThetaVault = artifacts.require("RibbonThetaVault");
+const AdminUpgradeabilityProxy = artifacts.require("AdminUpgradeabilityProxy");
+const ProtocolAdapterLib = artifacts.require("ProtocolAdapter");
+const { encodeCall } = require("@openzeppelin/upgrades");
+const { ethers, BigNumber } = require("ethers");
+const { parseEther } = ethers.utils;
+
+const {
+  updateDeployedAddresses,
+} = require("../scripts/helpers/updateDeployedAddresses");
+const ACCOUNTS = require("../constants/accounts.json");
+const DEPLOYMENTS = require("../constants/deployments.json");
+const EXTERNAL_ADDRESSES = require("../constants/externalAddresses.json");
+
+module.exports = async function (deployer, network) {
+  const networkLookup = network.replace("-fork", "");
+  const { admin, owner } = ACCOUNTS[networkLookup];
+
+  await ProtocolAdapterLib.deployed();
+
+  await deployer.link(ProtocolAdapterLib, RibbonThetaVault);
+
+  // Deploying the logic contract
+  await deployer.deploy(
+    RibbonThetaVault,
+    EXTERNAL_ADDRESSES[networkLookup].assets.weth,
+    DEPLOYMENTS[networkLookup].RibbonFactory,
+    EXTERNAL_ADDRESSES[networkLookup].assets.weth,
+    EXTERNAL_ADDRESSES[networkLookup].assets.usdc,
+    EXTERNAL_ADDRESSES[networkLookup].yearnRegistry,
+    EXTERNAL_ADDRESSES[networkLookup].airswapSwap,
+    18,
+    // WETH: 10**18, 10**10 0.0000001
+    // WBTC: 0.000001
+    BigNumber.from("10").pow(BigNumber.from("10")).toString(), // WBTC 10**3
+    false,
+    true,
+    { from: admin }
+  );
+  await updateDeployedAddresses(
+    network,
+    "RibbonyvETHCoveredCallLogic",
+    RibbonThetaVault.address
+  );
+
+  // Deploying the proxy contract
+  const initBytes = encodeCall(
+    "initialize",
+    ["address", "address", "uint256", "string", "string"],
+    [
+      owner,
+      owner,
+      parseEther("1000").toString(),
+      "Ribbon yvETH Theta Vault",
+      "rETH-THETA-YEARN",
```

Comment:
initially went with ryvETH-THETA but don't want a misleading name - shares minted will be based on `asset` deposited, not on `collateral asset` deposited. 

ex: deposit 1eth, will get ~1 rtoken. not deposit 1eth -> 0.9yvweth -> ~0.9 rtoken.